docs/content/configuration.md

```diff
@@ -925,3 +929,13 @@ with data put into the configured `directory`.
 | `storage.disk.badger` | `string` | No (default: empty) | "Superflags" passed to Badger allowing to modify advanced options. |
 
 See [the docs on disk storage](../misc-disk/) for details about the settings.
+
+### Server
+
+The `server` configuration sets the gzip compression settings for `/v0/data`, `/v1/data` and `/v1/compile` HTTP `POST` endpoints
+The gzip compression settings are used when the client sends `Accept-Encoding: gzip`
+
+| Field                           | Type  | Required            | Description                                                                                                                                                                                                                                  |
+|---------------------------------|-------|---------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `server.gzip_min_length`        | `int` | No, (default: 1024) | Specifies the minimum length of the response to compress                                                                                                                                                                                     |
+| `server.gzip_compression_level` | `int` | No, (default: 9)    | Specifies the compression level. Accepted values: a value of either 0 (no compression), 1 (best speed, lowest compression) or 9 (slowest, best compression). See https://pkg.go.dev/compress/flate#pkg-constants |
```

Comment:
It's possible in the future we could support other encoding algorithms. So we could do something like `server.encoding.gzip.min_length`  and `server.encoding.gzip.compression_level`

Reply:
fixed, I've rewrote the config/parsing/documentation so we'll have `server.encoding.gzip`